File: frontend/src/lib/models/StationSummaryDTO.ts
import type { PositionDTO } from "./PositionDTO";

/**
 *
 * @export
 * @interface StationSummaryDTO
 */
export interface StationSummaryDTO {
	/**
	 *
	 * @type {PositionDTO}
	 * @memberof StationSummaryDTO
	 */
	position: PositionDTO;
	/**
	 *
	 * @type {string}
	 * @memberof StationSummaryDTO
	 */
	name: string;
	/**
	 *
	 * @type {number}
	 * @memberof StationSummaryDTO
	 */
	evaNumber: number;
}

File: backend/Navigator.Api/DTOs/Station/StationSummaryDTO.cs
using System.Text.Json.Serialization;

namespace Navigator.Api.DTOs.Station;

public class StationSummaryDTO
{
    [JsonPropertyName("evaNumber")]
    public required int EvaNumber { get; set; }

    [JsonPropertyName("name")]
    public required string Name { get; set; }

    [JsonPropertyName("position")]
    public required PositionDTO Position { get; set; }
}
